Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -4296,9 +4295,6 @@ void CGame::PlayerCompleteConnect(CPlayer* pPlayer)
 
     // Send him the join details
     pPlayer->Send(CPlayerConnectCompletePacket());
-
-    // The player is spawned when he's connected, just the Camera is not faded in/not targetting
-    pPlayer->SetSpawned(true);
```

Comment:
You have replaced IsSpawned with !IsDead(). But at this line you can see that they are not interchangeable. Because m_bIsDead is only set when player is wasted. But m_bSpawned is true when player's connection is completed, not only when he's spawned via spawnPlayer.

Reply:
that line itself doesnt make sense. When player joins a server (lets say without any resources) then he only joined but he is not spawned.

Reply:
Okay. But it should be noted that these changes break the backward compatibility. Lets take a look at `setElementHealth` for example. Currently `setElementHealth` is valid even if it's called before spawnPlayer. After your changes it will not be true anymore.

Reply:
On the other hand, who the hell would want to use setElementHealth if player is not even spawned, right?

Reply:
> On the other hand, who the hell would want to use setElementHealth if player is not even spawned, right?

It also *might potentially* fix #472 but I really doubt it.

Backwards compatibility wouldn't really be a problem here. A quick refactor of `setElementHealth` to return `false` when the player is not spawned

Reply:
I've unresolved this review because as it stands, it looks like the backwards compatibility concern hasn't been tackled.
`A quick refactor of setElementHealth to return false when the player is not spawned` (@TracerDS)
Then please open a PR to do that, which will be a blocker for this to resolve and merge.

Reply:
@TracerDS I see you're updating the PR, but not acknownledging the above review.
Can you clarify why, because otherwise merge isn't gonna happen.

Reply:
> @TracerDS I see you're updating the PR, but not acknownledging the above review. Can you clarify why, because otherwise merge isn't gonna happen.

`setElementHealth` currently checks if player was spawned.
The value of `IsSpawned()` is the same as `!IsDead()`.
They were interchangeable. `m_bIsDead` is set to `false` in `CPed`'s constructor.
`m_bIsSpawned` is set to `true` **only** here.
So in current mta's code its always set to `true`.

In this PR you cannot set player's health when he is not connected to the server yet.
If it stayed the same then it might cause some issues.
Checking against `IsDead` instead fixes these issues and returns false when ped is dead/not spawned.
However, that would also mean that using `setElementHealth` on a dead ped will fail (return false). If that is a concern, I could remove the whole `if` statement here since its always set to `true` in previous code